datastore/localjson.go

```diff
@@ -7,128 +7,81 @@ import (
 	"log/slog"
 	"os"
 	"path/filepath"
+	"reflect"
 	"time"
 
-	"github.com/Santiago-Labs/go-ocsf/ocsf"
 	"github.com/samsarahq/go/oops"
 )
 
-type localJsonDatastore struct {
-	BaseDatastore
+type localJsonDatastore[T any] struct {
+	BaseDatastore[T]
 
-	currentFindingsPath   string
-	currentActivitiesPath string
+	currentPath string
+	basepath    string
 }
 
 // localJsonDatastore implements the Datastore interface using local JSON files for storage.
-// It provides methods to retrieve, save, and manage vulnerability findings in JSON format.
-func NewLocalJsonDatastore(ctx context.Context) (Datastore, error) {
-	if err := os.MkdirAll(BasepathFindings, 0755); err != nil {
-		return nil, oops.Wrapf(err, "failed to create directory")
-	}
+// It provides methods to retrieve, save, and manage ocsf data in JSON format.
+func NewLocalJsonDatastore[T any](ctx context.Context) (Datastore[T], error) {
 
-	if err := os.MkdirAll(BasepathActivities, 0755); err != nil {
+	typeName := reflect.TypeOf((*T)(nil)).Elem().Name()
+	if err := os.MkdirAll(basepaths[typeName], 0755); err != nil {
 		return nil, oops.Wrapf(err, "failed to create directory")
 	}
 
-	s := &localJsonDatastore{}
+	s := &localJsonDatastore[T]{
+		basepath: basepaths[typeName],
+	}
 
-	s.BaseDatastore = BaseDatastore{
+	s.BaseDatastore = BaseDatastore[T]{
 		store: s,
 	}
 
 	return s, nil
 }
 
-// GetFindingsFromFile retrieves all vulnerability findings from a specific file path.
-// It reads the gzipped JSON file and parses it into a slice of vulnerability findings.
-func (s *localJsonDatastore) GetFindingsFromFile(ctx context.Context, path string) ([]ocsf.VulnerabilityFinding, error) {
+// GetItemsFromFile retrieves all ocsf data from a specific file path.
+// It reads the gzipped JSON file and parses it into a slice of ocsf data.
+func (s *localJsonDatastore[T]) GetItemsFile(ctx context.Context, path string) ([]T, error) {
```

Comment:
```suggestion
func (s *localJsonDatastore[T]) GetItemsFromFile(ctx context.Context, path string) ([]T, error) {
```
so it matches the name in the comment and because I think its more clear